API.php

```diff
@@ -1058,6 +1059,7 @@ public function updateContainerVersion($idSite, $idContainer, $idContainerVersio
     {
         $name = $this->decodeQuotes($name);
         $this->accessValidator->checkWriteCapability($idSite);
+        $this->accessValidator->checkUseCustomTemplatesCapability($idSite);
```

Comment:
@AltamashShaikh 

Question (potentially blocking): When I go to the "Versions" page on my write access user, on 5.x-dev it would let me edit the "Version name" or "Version description". With this new line of code, I now get a generic error in the UI. Behind the scenes it logs as:

ERROR API[2024-05-06 08:26:19 UTC] [c5eb8] Uncaught exception in API: You can't access this resource as it requires 'tagmanager_use_custom_templates' capability for the website id = 1. /Users/ajr/repos/matomo/core/Access.php:760

While it may be correct that we want to block the Write user from editing these Version details, should it at least display a well-defined error? Perhaps block it in the Vue file?

    updateVersion() {
      ...
      if (!this.hasPublishCapability()) {
        this.showNotification('Specific error', 'error');
        return;
      }

![image](https://github.com/matomo-org/tag-manager/assets/1289711/830fcc9b-650b-420d-a8c1-479ec87a3ef7)